kotlinx-coroutines-core/jvm/test/sync/SemaphoreStressTest.kt

```diff
@@ -57,4 +65,36 @@ class SemaphoreStressTest : TestBase() {
         semaphore.release()
         assertEquals(1, semaphore.availablePermits)
     }
-}
\ No newline at end of file
+
+    @Test
+    fun stressReleaseCancellation() = runTest {
+        val n = 10_000 * stressTestMultiplier
+        val semaphore = Semaphore(1, 1)
+        repeat (n) {
+            assertEquals(0, semaphore.availablePermits)
+
+            var shared = false
+
+            val job1 = launch(pool) {
```

Comment:
A comment would be helpful here, explaining what is being tested and why we need a single-threaded pool.